File: app/src/main/java/ch/bailu/aat/views/TagEditor.java
package ch.bailu.aat.views;

import android.content.Context;
import android.text.InputType;
import android.widget.EditText;

import ch.bailu.aat.util.TextBackup;
import ch.bailu.aat.util.ui.AppLog;
import ch.bailu.util_java.foc.Foc;


public class TagEditor extends EditText {
    private Foc backup;


    public TagEditor(Context context, Foc an)  {
        super(context);

        createEditor();
        createEditorBackup(an);
    }

    private void createEditorBackup(Foc directory) {
        backup = directory.child("edit.txt");
        showFile();
    }


    private void createEditor() {
        setInputType(InputType.TYPE_TEXT_FLAG_MULTI_LINE
                | InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS
                | InputType.TYPE_CLASS_TEXT
                | InputType.TYPE_TEXT_FLAG_IME_MULTI_LINE);
        setHorizontallyScrolling(true);
    }



    @Override
    public void onDetachedFromWindow() {
        saveFile();
        super.onDetachedFromWindow();
    }


    private void showFile() {
        String text;
        try {
            text = TextBackup.read(backup);

        } catch (Exception e) {
            text = "";
        }
        setText(text);

    }


    private void saveFile() {
        try {
            TextBackup.write(backup, getText().toString());
        } catch (Exception e) {
            AppLog.e(getContext(), TagEditor.this, e);

        }
    }


}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/osm/TagEditor.kt
package ch.bailu.aat.views.osm

import android.content.Context
import android.text.InputType
import android.widget.EditText
import ch.bailu.aat.util.fs.TextBackup
import ch.bailu.aat_lib.logger.AppLog
import ch.bailu.aat_lib.util.fs.AppDirectory
import ch.bailu.foc.Foc

class TagEditor(context: Context, an: Foc) : EditText(context) {
    private var backup: Foc = Foc.FOC_NULL

    init {
        createEditor()
        createEditorBackup(an)
    }

    private fun createEditorBackup(directory: Foc) {
        backup = directory.child(AppDirectory.FILE_EDIT_BACKUP)
        showFile()
    }

    private fun createEditor() {
        inputType = (InputType.TYPE_TEXT_FLAG_MULTI_LINE
                or InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS
                or InputType.TYPE_CLASS_TEXT
                or InputType.TYPE_TEXT_FLAG_IME_MULTI_LINE)
        setHorizontallyScrolling(true)
    }

    public override fun onDetachedFromWindow() {
        saveFile()
        super.onDetachedFromWindow()
    }

    private fun showFile() {
        val text = try {
            TextBackup.read(backup)
        } catch (e: Exception) {
            ""
        }
        setText(text)
    }

    private fun saveFile() {
        try {
            TextBackup.write(backup, text.toString())
        } catch (e: Exception) {
            AppLog.e(this@TagEditor, e)
        }
    }
}
